core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/move.kt

```diff
@@ -23,20 +24,70 @@ internal fun <T, C> MoveClause<T, C>.afterOrBefore(column: ColumnSelector<T, *>,
     val removeResult = df.removeImpl(columns = columns)
 
     val targetPath = df.getColumnWithPath(column).path
+    val sourcePaths = removeResult.removedColumns.map { it.toColumnWithPath<C>().path }
+
+    // Check if any source path is a prefix of the target path
+    sourcePaths.forEach { sourcePath ->
+        val sourceSegments = sourcePath.toList()
+        val targetSegments = targetPath.toList()
+
+        if (sourceSegments.size <= targetSegments.size &&
+            sourceSegments.indices.all { targetSegments[it] == sourceSegments[it] }
+        ) {
+            throw IllegalArgumentException(
+                "Cannot move column '${sourcePath.joinToString()}' after its child column '${targetPath.joinToString()}'",
+            )
+        }
+    }
+
     val removeRoot = removeResult.removedColumns.first().getRoot()
 
     val refNode = removeRoot.getOrPut(targetPath) {
-        val parentPath = it.dropLast(1)
-        val parent = if (parentPath.isEmpty()) df else df.getColumnGroup(parentPath)
-        val index = parent.getColumnIndex(it.last())
-        val col = df.getColumn(index)
+        val path = it.toList()
+
+        // Find the group reference (if any) and adjust the path
+        val groupRefIndex = path.indexOfFirst { it.isEmpty() }
```

Comment:
What is happening here? How does the first empty path piece correspond to a group node? And at which indices can this empty path piece occur?